From e1bb639db4ddc60403acee62f9f52c30ce399e60 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Tue, 25 Feb 2014 17:20:37 +0200
Subject: [PATCH 0012/1240] pxp: adjustments to run u-boot on palladium model

	- Disabled GIC initialization
	- Stopped CPU at reset vector
	- Changed DRAM base address

Change-Id: Iadbb830bf00e1f1e1b98ccffda7f00a08d8abbd8
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/cpu/armv8/start.S       | 1 +
 include/configs/armada8021-pxp.h | 6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/cpu/armv8/start.S b/arch/arm/cpu/armv8/start.S
index 4b11aa4..6b7aad9 100644
--- a/arch/arm/cpu/armv8/start.S
+++ b/arch/arm/cpu/armv8/start.S
@@ -20,6 +20,7 @@
 
 .globl	_start
 _start:
+	b .
 	b	reset
 
 	.align 3
diff --git a/include/configs/armada8021-pxp.h b/include/configs/armada8021-pxp.h
index 2c0196b..9b1e7f2 100644
--- a/include/configs/armada8021-pxp.h
+++ b/include/configs/armada8021-pxp.h
@@ -45,7 +45,7 @@
 #define CONFIG_BOOTP_VCI_STRING		"U-boot.armv8.armada8021-pxp"
 
 /* Link Definitions */
-#define CONFIG_SYS_TEXT_BASE		0x80000000
+#define CONFIG_SYS_TEXT_BASE		0x00000000
 #define CONFIG_SYS_INIT_SP_ADDR         (CONFIG_SYS_SDRAM_BASE + 0x7fff0)
 
 /* Flat Device Tree Definitions */
@@ -69,7 +69,7 @@
 #define V2M_SYSCTL			(V2M_PA_CS3 + V2M_PERIPH_OFFSET(2))
 #define V2M_SERIAL_BUS_PCI		(V2M_PA_CS3 + V2M_PERIPH_OFFSET(3))
 
-#define V2M_BASE			0x80000000
+#define V2M_BASE			0x00000000
 
 /*
  * Physical addresses, offset from V2M_PA_CS0-3
@@ -171,7 +171,7 @@
 /* Physical Memory Map */
 #define CONFIG_NR_DRAM_BANKS		1
 #define PHYS_SDRAM_1			(V2M_BASE)	/* SDRAM Bank #1 */
-#define PHYS_SDRAM_1_SIZE		0x80000000	/* 2048 MB */
+#define PHYS_SDRAM_1_SIZE		0x20000000	/* 2048 MB */
 #define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
 
 /* Initial environment variables */
-- 
1.9.1

